supervisor/process.py

```diff
@@ -549,6 +549,14 @@ def finish(self, pid, sts):
             msg = "stopped: %s (%s)" % (processname, msg)
             self._assertInState(ProcessStates.STOPPING)
             self.change_state(ProcessStates.STOPPED)
+            if exit_expected:
+               self.config.options.logger.info(msg)
+            else:
+               if "SIGTERM" in msg or "SIGKILL" in msg:
```

Comment:
No, I don't think we can do this.  

In the config file, the user may set `stopsignal = TERM` in a `[program:x]` and then `TERM` is the normal signal that is sent and is no problem.  This should not be logged as a warning then.

Also, I don't think we should scrape the log message string to determine if a problem occurred when then information should already be available in proper data structures.

Reply:
How about we have two log levels: INFO and WARN. 
if exit_expected:
               self.config.options.logger.info(msg)
else:
               self.config.options.logger.warn(msg)

Reply:
> How about we have two log levels: INFO and WARN.
> ```
> if exit_expected:
>   self.config.options.logger.info(msg)
> else:
>   self.config.options.logger.warn(msg)
>```

👍 I would merge that.

Reply:
I just want to make sure I cover everything.
How about a process core dumped case? It will fell into else case. Is CRIT log level suitable for core dumped case? 

if self.killing:
   if exit_expected:
         self.config.options.logger.info(msg)
   else:
         self.config.options.logger.warn(msg)
elif too_quickly:      
        self.config.options.logger.warn(msg)
else: 
   if exit_expected:
         self.config.options.logger.info(msg)
   else:
        if sts & 0x80:
            self.config.options.logger.critical(msg)
        else:
            self.config.options.logger.warn(msg)

Log:
Dec  1 01:25:16.787565 2020 sonic INFO swss#supervisord 2020-12-01 01:25:15,377 CRIT exited: nbrmgrd (terminated by SIGQUIT (core dumped); not expected)

Reply:
> ```
> if self.killing:
>      if exit_expected:
>          self.config.options.logger.info(msg)
>      else:
>          self.config.options.logger.warn(msg)
> ```

:+1:

> ```
> elif too_quickly:        
>      self.config.options.logger.warn(msg)
> ```

:+1: 

> ```
> else: 
>      if exit_expected:
>            self.config.options.logger.info(msg)
>      else:
>            if sts & 0x80:
>                  self.config.options.logger.critical(msg)
>            else:
>                  self.config.options.logger.warn(msg)
> ```

As far as I can tell with `grep`, the only places `supervisord` uses `critical` are for problems with itself (not subprocesses).  For that reason, I am inclined to not use `critical`.  

Could the simpler rule of just "expected→info, unexpected→warn" be sufficient?  That would be easier to document and understand.

Reply:
Thank you for the clarification.
  
Is it possible to have the log severity level of core dumped processes to be configurable (the default is WARN) in the supervisord configuration file?